test/Microsoft.Azure.Devices.Edge.Test/helpers/SasManualProvisioningFixture.cs

```diff
@@ -23,19 +21,19 @@ protected virtual async Task SasProvisionEdgeAsync(bool withCerts = false)
                 CancellationToken token = cts.Token;
                 DateTime startTime = DateTime.Now;
 
-                EdgeDevice device = await EdgeDevice.GetOrCreateIdentityAsync(
-                    Context.Current.DeviceId.GetOrElse(DeviceId.Current.Generate()),
-                    Context.Current.ParentDeviceId,
+                this.device = await EdgeDevice.GetOrCreateIdentityAsync(
+                    DeviceId.Current.Generate(),
```

Comment:
>DeviceId.Current.Generate(), [](start = 20, length = 28)

this looks like we won't ever use the DeviceId from the context anymore, is that intentional?

Reply:
We are still doing the same thing when the `NestedEdgeConfig(5)` is called here :
https://github.com/Azure/iotedge/blob/2df777c5aa1eff2c9c1ada027920f648dd610c02/test/Microsoft.Azure.Devices.Edge.Test/helpers/ManualProvisioningFixture.cs#L77

I'm basically consolidate the parameter checking logic to the same spot in `NestedEdgeConfig` and then read-only throughout the test cases (instead of read & check them every test case).